### PR TITLE
Fixes to MSVC support for ci branch.

### PR DESCRIPTION
**Pull request by Rossmaxx**
I have been helping @JohannesLorenz  with the ci branch for some time and have some fixes now. Here they are.

Note : To build on msvc properly, we should wait till liblo gets an updated release (or the workaround I used, which is to use `--head` while installing liblo on vcpkg. Though this step needs some debugging).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,7 +26,10 @@ else()
 endif(FLTK_FOUND)
 
 if(MSVC)
-    find_package(LIBLO)
+    find_package(liblo)
```
Comment (JohannesLorenz) [inline CMakeLists.txt:29]: What happened here? Did `find_package(LIBLO)` not work?

File: CMakeLists.txt [hunk 2]
```diff
@@ -185,8 +188,7 @@ maketest(message-alignment)
 maketest(test-arg-iter)
 if(LIBLO_FOUND)
     maketest(liblo)
-    target_include_directories(liblo PRIVATE ${LIBLO_INCLUDE_DIRS})
```
Comment (JohannesLorenz) [inline CMakeLists.txt:188]: Why removing the include directory?

File: CMakeLists.txt [hunk 3]
```diff
@@ -206,9 +208,7 @@ maketestcpp(path-search)
 
 maketestcpp(performance)
 if(LIBLO_FOUND)
-    target_include_directories(performance PRIVATE ${LIBLO_INCLUDE_DIRS})
-    target_link_libraries(performance      PRIVATE ${LIBLO_LIBRARIES})
-    target_compile_definitions(performance PRIVATE HAVE_LIBLO)
```
Comment (JohannesLorenz) [inline CMakeLists.txt:211]: Why did you remove the `HAVE_LIBLO` define? Wouldn't this mean that `performance.cpp` compiles without liblo?

**Review by JohannesLorenz — COMMENTED** [review]
Thanks for your submission. I added 3 comments with questions.

File: CMakeLists.txt (hunk 1 above)
Comment (Rossmaxx) [reply CMakeLists.txt:29]: That one points to our custom findliblo, which was renoved. The one i added points to findliblo which is added within liblo's cmake integration.

File: CMakeLists.txt (hunk 2 above)
Comment (Rossmaxx) [reply CMakeLists.txt:188]: The include dirs are being found by liblo in their master branch's integration. Also, the LIBLO_INCLUDE_DIRS variable is not working anymore because we are not having the custom findliblo.cmake

File: CMakeLists.txt (hunk 3 above)
Comment (Rossmaxx) [reply CMakeLists.txt:211]: Oops, mb. Should have noticed that one. Will fix now

File: CMakeLists.txt (hunk 1 above)
Comment (JohannesLorenz) [reply CMakeLists.txt:29]: Oh, so `cmake`'s integration requires it in small letters?

File: CMakeLists.txt (hunk 2 above)
Comment (JohannesLorenz) [reply CMakeLists.txt:188]: So the CMake's FindLiblo does not set `LIBLO_INCLUDE_DIRS`?

**Comment by fundamental** [comment]
To move this PR forward, could there be a rebase onto our main branch to eliminate conflicts when this is merged and then a bit of git rebase magic to combine/fixup WIP/testing commits?

**Comment by JohannesLorenz** [comment]
@Rossmaxx I am still working with the liblo team to setup a new release. In the mean time, can you please fix at least the issues which are still open?

@fundamental That is what we usually do at the end. Right now, I do not want to touch it because it is already broken and I do not want to add more uncertainty to this PR. When the PR works, it should be the right time for a rebase.

File: CMakeLists.txt (hunk 3 above)
Comment (Rossmaxx) [reply CMakeLists.txt:211]: Forgot about this pr. Will get back to this soon.

File: CMakeLists.txt (hunk 1 above)
Comment (Rossmaxx) [reply CMakeLists.txt:29]: Basically yeah.

**Comment by Rossmaxx** [comment]
> To move this PR forward, could there be a rebase onto our main branch to eliminate conflicts when this is merged and then a bit of git rebase magic to combine/fixup WIP/testing commits?

It would be of little use as this pr is not needed for master in any way and this only affects the ci branch. Merging ci with master can fix this.

If you are talking about squashing the commits, i would like to but don't know much about that. I remember messing up a rebase one time and lost all my commits.

File: CMakeLists.txt (hunk 3 above)
Comment (Rossmaxx) [reply CMakeLists.txt:211]: In my latest commit I pushed the performance test into liblo_found, this will do the same thing I guess. I'm not resolving this for now so as to keep my approach open for discussion.

Comment (Rossmaxx) [reply CMakeLists.txt:211]: I misunderstood the comment. I fixed it now.

**Comment by JohannesLorenz** [comment]
Thanks @Rossmaxx . I will merge this.

**Comment by JohannesLorenz** [comment]
Oh, sorry, I cannot merge. Can you please solve the conflicts?

**Comment by Rossmaxx** [comment]
> Oh, sorry, I cannot merge. Can you please solve the conflicts?

There's none for ci branch. You might be trying to merge master.

Btw did we get a new liblo release?

**Comment by JohannesLorenz** [comment]
Ah right, I could create a merge commit, only rebase was not possible.

For liblo, we were working on a new release until today. There might be a release in this month, possibly.

**Comment by Rossmaxx** [comment]
> For liblo, we were working on a new release until today. There might be a release in this month, possibly.

Notify me when it's done so that i can open a pr to add liblo's new version to vcpkg.
